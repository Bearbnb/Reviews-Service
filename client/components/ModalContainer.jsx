import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import ModalButton from './ModalButton.jsx';
import styles from '../../styles/FlagModal.css';


const ModalContainer = ({ hideModalFunc, checkedRadio, changeRadio }) => (
  <div>
    <div className={styles.modalOverlay} onClick={() => hideModalFunc()} />
    <div className={styles.modal}>
      <div className={styles.modalContent}>
        <div className={styles.header}>
          <button
            type="button"
            onClick={() => hideModalFunc()}
          >
            <i className="fas fa-times" />
          </button>
          <h1>Do you want to anonymously report this review?</h1>
        </div>
        <div className={styles.FlagText}>
            If so, please choose one of the following reasons
        </div>
        <div className={styles.FlagButtons}>
          <ModalButton
            header="Inappropriate Content"
            content="This review contains violent, graphic, promotional, or otherwise offensive content."
            changeRadio={changeRadio}
          />
          <ModalButton
            header="Dishonest or hateful content"
            content="This review is purposefully malicious and assaulting."
            changeRadio={changeRadio}
          />
          <ModalButton 
          header="Fake content" 
          content="This review contains false information or may be fake." 
          changeRadio={changeRadio}
          />
        </div>
        <button
          type="button"
          className={checkedRadio ? styles.FlagSubmit : styles.FlagSubmitHide}
        >Submit
        </button>
      </div>
    </div>


  </div>
);

export default ModalContainer;
