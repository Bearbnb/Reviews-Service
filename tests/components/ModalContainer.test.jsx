import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ModalContainer from '../../client/components/ModalContainer';

describe('<ModalContainer />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<ModalContainer
      hideModalFunc={jest.fn()}
      checkedRadio={false}
      changeRadio={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should hide modal when clicking outer div', () => {
    const hideModalFunc = jest.fn();
    const wrapper = shallow(<ModalContainer
      hideModalFunc={hideModalFunc}
      checkedRadio={false}
      changeRadio={jest.fn()}
    />);

    wrapper.find('div div').first().simulate('click');
    expect(hideModalFunc).toHaveBeenCalled();
  });

  it('should hide modal when clicking X', () => {
    const hideModalFunc = jest.fn();
    const wrapper = shallow(<ModalContainer
      hideModalFunc={hideModalFunc}
      checkedRadio={false}
      changeRadio={jest.fn()}
    />);
    wrapper.find('div > div > div > div > button').first().simulate('click');
    expect(hideModalFunc).toHaveBeenCalled();
  });

  it('should have 3 Modal Buttons', () => {
    const hideModalFunc = jest.fn();
    const wrapper = shallow(<ModalContainer
      hideModalFunc={hideModalFunc}
      checkedRadio={false}
      changeRadio={jest.fn()}
    />);
    const parentButtons = wrapper.find('ModalButton');
    expect(parentButtons.length).toBe(3);
  });
});
