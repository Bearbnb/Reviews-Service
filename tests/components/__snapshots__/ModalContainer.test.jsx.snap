// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalContainer /> should render properly 1`] = `
<div>
  <div
    className="modalOverlay"
    onClick={[Function]}
    role="presentation"
  />
  <div
    className="modal"
  >
    <div
      className="modalContent"
    >
      <div
        className="header"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          <i
            className="fas fa-times"
          />
        </button>
        <h1>
          Do you want to anonymously report this review?
        </h1>
      </div>
      <div
        className="FlagText"
      >
        If so, please choose one of the following reasons
      </div>
      <div
        className="FlagButtons"
      >
        <ModalButton
          changeRadio={[MockFunction]}
          content="This review contains violent, graphic, promotional, or otherwise offensive content."
          header="Inappropriate Content"
        />
        <ModalButton
          changeRadio={[MockFunction]}
          content="This review is purposefully malicious and assaulting."
          header="Dishonest or hateful content"
        />
        <ModalButton
          changeRadio={[MockFunction]}
          content="This review contains false information or may be fake."
          header="Fake content"
        />
      </div>
      <button
        className="FlagSubmitHide"
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
